engines/ceg_prophecy_arrays.cpp

```diff
@@ -142,7 +142,7 @@ ProverResult CegProphecyArrays<Prover_T>::check_until(int k)
 
       if (res == ProverResult::FALSE) {
         // use witness length
-        reached_k_ = prover->witness_length() - 1;
+        reached_k_ = prover->witness_length();
```

Comment:
I am thinking that it should be `witness_length() -1`

Reply:
because reached_k_ means that there is no bug up to that bound

Reply:
maybe we have to make the change elsewhere

Reply:
Oh interesting. Right. So I was referring to the witness length default implementation when talking about `reached_k_`. I hadn’t thought about this one. That does make sense that this should also subtract 1 because `reached_k_` should always be the last bound finished. I’ll make that change and make sure that `reached_k_` is used correctly for the refinement step.

Reply:
All right, it already worked -- good point. Made with 693002c7c7d6bf7ac57cbf45d3417e7c718a30be.